File: Easy/Hash Table/205. Isomorphic Strings.py
# 205. Isomorphic Strings.cpp

class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:

        st = set()
        mp = {}

        n = len(s)

        for i in range(0,n,1):
            if s[i] in mp:
                if mp[s[i]] != t[i] : return False
            
            else: 
                if t[i] in st : return False
                mp[s[i]] = t[i]
                st.add(t[i])
        
        return True
        

File: Easy/Hash Table/205. Isomorphic Strings.cpp
//205. Isomorphic Strings ->c++

class Solution {
public:
    bool isIsomorphic(string s, string t) {

        unordered_map<char,char>mp_s;
        unordered_set<char>st;
        int n = s.size();

        for(int i = 0; i< n; i++){

            if(mp_s.find(s[i]) != mp_s.end()){
                if(mp_s[s[i]] != t[i]) return false;
            }
            else{
                if(st.find(t[i]) != st.end()) return false;

                st.insert(t[i]);
                
                 mp_s[s[i]] = t[i];

            }
            
        }

        return true;
        
    }
};

